src/jep/InvocationHandler.java

```diff
@@ -53,11 +55,12 @@
      * @exception JepException
      *                if an error occurs
      */
-    public InvocationHandler(long tstate, long ltarget, Jep jep)
+    public InvocationHandler(long tstate, long ltarget, Jep jep, final boolean functionalInterface)
```

Comment:
We should keep the old method signature for backwards compatibility.

```java
public InvocationHandler(long tstate, long ltarget, Jep jep) {
    this(tstate, ltarget, jep, false);
}
```
Honestly @mrj0 is most familiar with the `Proxy` and `InvocationHandler` code.  Do people new up an `InvocationHandler` or just use `Proxy.newProxyInstance(...)`?  Either we should keep the backwards compatibility as specified above or we should make the constructor protected if it's not supposed to be public API.